restorer.go

```diff
@@ -44,8 +45,20 @@ func (res *Restorer) restoreTo(dst string, dir string, treeID backend.ID) error
 	}
 
 	for _, node := range tree.Nodes {
-		if err := res.restoreNodeTo(node, dir, dst); err != nil {
-			return err
+		selectedForRestore := true
+		if res.SelectForRestore != nil {
```

Comment:
Hm, I like that. I suppose this doesn't really introduce any performance problems.